rotkehlchen/db/dbhandler.py

```diff
@@ -662,24 +662,24 @@ def remove_from_ignored_assets(self, write_cursor: 'DBCursor', asset: Asset) ->
             (asset.identifier,),
         )
 
-    # pylint: disable=no-self-use
-    def get_ignored_assets(self, write_cursor: 'DBCursor') -> List[Asset]:
-        write_cursor.execute(
+    def get_ignored_assets(self, cursor: 'DBCursor') -> List[Asset]:
+        cursor.execute(
             'SELECT value FROM multisettings WHERE name="ignored_asset";',
         )
         assets = []
-        for asset_setting in write_cursor:
+        for asset_setting in cursor:
             try:
                 asset = Asset(asset_setting[0])
             except UnknownAsset:
                 msg = (
                     f'Found unknown asset {asset_setting[0]} in the list of ignored '
                     f'assets. Removing it.'
                 )
-                write_cursor.execute(
-                    'DELETE FROM multisettings WHERE name="ignored_asset" AND value=?;',
-                    (asset_setting[0],),
-                )
+                with self.user_write() as write_cursor:
+                    write_cursor.execute(
+                        'DELETE FROM multisettings WHERE name="ignored_asset" AND value=?;',
+                        (asset_setting[0],),
+                    )
```

Comment:
This area here is the actual bugfix. The same write cursor was being iterated and then during iteration (in the case of an UnknownAsset exception) modified and then  attempted to continue the iteration.